### PR TITLE
Checking for url existence in possible new default shop

### LINKED ISSUE #23753: As a user, I shouldn't be able to select a shop as default if it doesn't have an URL configured.
#### Describe the bug

In the BO -> Advanced Parameters -> Multistore page -> when we set a shop (without main URL) as a default shop, then if we disable the multistore, the FO can't be displayed.

#### Steps to Reproduce

1. Go to BO > Shop Parameters > General page & Enable multistore 
2. Go to BO -> Advanced Parameters -> Multistore page 
3. Create a new shop without adding an URL
4. In the AdminShopGroup controller > Multistore Option, select the shop without URL & save
5. Go to BO > Shop Parameters > General page & disable the multistore
6. Click View my shop
7. See error:  FO can't be opened because there is no URL for this shop

#### Screenrecord

https://drive.google.com/file/d/1iGV2jzXKlZSmHeSWsQl2clu4KrhhTNNa/view?usp=sharing

#### Wanted behavior

As a user, I shouldn't be able to select a shop as default if it doesn't have an URL configured.

#### Specs
In the AdminShopGroup controller, if I select a shop as default and if it doesn't have an URL configured, when I click on "Save" an error message is displayed "You must configure the URL of this store before you can set it as default." in _Admin.Advparameters.Notification_ and the shop selected isn't defined as default.

#### Additional information

* PrestaShop version: 1.7.7.3build1, 1769
* PHP version: 7.2

### PR DESCRIPTION
**Pull request by kpodemski**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | title says it all
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #23753
| How to test?      | Follow instructions from #23753
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.
| Sponsor company | impSolutions

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24796)

### CODE REVIEW

File: controllers/admin/AdminShopGroupController.php [hunk 1]
```diff
@@ -344,6 +344,13 @@ public function postProcess()
         return parent::postProcess();
     }
 
+    public function beforeUpdateOptions()
+    {
+        if (!(new Shop((int) Tools::getValue('PS_SHOP_DEFAULT')))->getBaseURL()) {
+            $this->errors[] = $this->trans('You must configure the URL of this store before you can set it as default.', [], 'Admin.Advparameters.Notification');
```
Comment (Julievrz) [inline controllers/admin/AdminShopGroupController.php]:
```suggestion
            $this->errors[] = $this->trans('You must configure this store's URL before setting it as default.', [], 'Admin.Advparameters.Notification');
```

**Comment by Julievrz** [comment]
Good catch, thank you @kpodemski :-)

**Review by Progi1984 — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Comment by florine2623** [comment]
LGTM

Thanks @kpodemski !

**Comment by Progi1984** [comment]
Thanks @kpodemski & @florine2623

**Pull request merged**